src/openrct2/ride/Vehicle.cpp

```diff
@@ -4576,12 +4575,12 @@ static void vehicle_update_boat_location(Vehicle* vehicle)
     TileCoordsXY returnPosition = ride->boat_hire_return_position;
     uint8_t returnDirection = ride->boat_hire_return_direction & 3;
 
-    TileCoordsXY location{ CoordsXY{ vehicle->x, vehicle->y } + CoordsDirectionDelta[returnDirection] };
+    CoordsXY location = CoordsXY{ vehicle->x, vehicle->y } + CoordsDirectionDelta[returnDirection];
 
-    if (location == returnPosition)
+    if (location == returnPosition.ToCoordsXY().ToTileStart())
```

Comment:
location should be at tile start. return position is fine. This will cause issues with pathfinding to exits.